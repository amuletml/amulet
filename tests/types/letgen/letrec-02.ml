let rec (f, g) = (foo, bar)
and foo = f
and bar = g
