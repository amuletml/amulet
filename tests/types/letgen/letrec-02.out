letrec-02.ml[1:5 ..3:7]: error
  Pattern bindings can not participate in recursion

  │ 
1 │ let (f, g) = (foo, bar)
  │     ^^^^^^

  Note: this binding is in the same recursive group as these

  │ 
2 │ and foo = f
  │     ^^^
3 │ and bar = g
  │     ^^^
